pkg/operators/localmanager/localmanager.go

```diff
@@ -273,21 +278,38 @@ func (l *localManagerTrace) PreGadgetRun() error {
 
 	// If --host is set, we do not want to create the below map because we do not
 	// want any filtering.
-	if setter, ok := l.gadgetInstance.(MountNsMapSetter); ok && !host {
-		// Create mount namespace map to filter by containers
-		mountnsmap, err := l.manager.igManager.CreateMountNsMap(l.subscriptionKey, containerSelector)
-		if err != nil {
-			return commonutils.WrapInErrManagerCreateMountNsMap(err)
-		}
+	if setter, ok := l.gadgetInstance.(MountNsMapSetter); ok {
+		if !host {
+			if l.manager.igManager == nil {
+				return fmt.Errorf("container-collection isn't available")
+			}
+
+			// Create mount namespace map to filter by containers
+			mountnsmap, err := l.manager.igManager.CreateMountNsMap(id.String(), containerSelector)
+			if err != nil {
+				return commonutils.WrapInErrManagerCreateMountNsMap(err)
+			}
 
-		log.Debugf("set mountnsmap for gadget")
-		setter.SetMountNsMap(mountnsmap)
+			log.Debugf("set mountnsmap for gadget")
+			setter.SetMountNsMap(mountnsmap)
 
-		l.mountnsmap = mountnsmap
+			l.mountnsmap = mountnsmap
+		} else if l.manager.igManager == nil {
+			log.Warn("container-collection isn't available no container filtering will be done")
```

Comment:
Mention enrichment won't work either:

```suggestion
			log.Warn("container-collection isn't available: container enrichment and filtering won't work")
```

Here or in the `Instantiate` when you set `enrichEvents`, it's the same.